src/mqtt/qmqtt_routesubscription.cpp

```diff
@@ -41,15 +42,28 @@ namespace QMQTT {
 
 Q_LOGGING_CATEGORY(routerSubscription, "qmqtt.routersubscription")
 
-RouteSubscription::RouteSubscription(Router *parent) : QObject(parent)
+RouteSubscription::RouteSubscription(Router *parent)
+    : QObject(parent)
+    , _client(parent->client())
 {
 }
 
+RouteSubscription::~RouteSubscription()
+{
+    unsubscribe();
+}
+
 QString RouteSubscription::route() const
 {
     return _topic;
 }
 
+void RouteSubscription::unsubscribe()
```

Comment:
`unsubscribe()` is a public method, also called in the destructor of `RouteSubscription`. Could multiple calls of `unsubscribe()` cause issues?